### PR TITLE
Modify/provisioner tweaks

### PR DESCRIPTION
**Pull request by tomjn**
This builds on the work in #2317 and extends the improvements across the main provisioners. It also introduces the `set -eo pipefail` in a lot of main provisioners, reduces output when it isn't needed, and fixes several installation issues related to composer and shyaml

Screenshot 2020-12-15 at 19 08 44
Screenshot 2020-12-15 at 19 08 58
Screenshot 2020-12-15 at 19 09 17

## Checks

* [x] I've updated the changelog.
* [x] I've tested this PR
* [x] This PR is for the `develop` branch not the `stable` branch.
* [x] This PR is complete and ready for review.

### CODE REVIEW

File: config/homebin/vagrant_provision [hunk 1]
```diff
@@ -65,6 +68,6 @@ else
 	echo -e "${RED} ⚠ The failed provisioners left these logs:"
 
 	# List failed provisioners, and pipe them through xargs to print them in a formatted way.
-	ls -A /vagrant/failed_provisioners/ | xargs -I "LOGFILE" echo -e "   - log/provisioners/${date_time}/LOGFILE.log"
+	ls -A /vagrant/failed_provisioners/ | xargs -I "LOGFILE" vvv_error "   - <b>log/provisioners/${date_time}/LOGFILE.log</b>"
```
Comment (Mte90) [inline config/homebin/vagrant_provision:71]: this is an html tag, why on a log file?

Comment (Mte90) [reply config/homebin/vagrant_provision:71]: Ok I saw in another file that will be replaced, maybe add something in our doc website?

Comment (tomjn) [reply config/homebin/vagrant_provision:71]: At the moment we don't document any of these functions so we'd want to figure out a solution for that, something like PHPDoc for shell scripts

Comment (Mte90) [reply config/homebin/vagrant_provision:71]: https://github.com/reconquest/shdoc There is this that is for bash, maybe is for another ticket?

Comment (tomjn) [reply config/homebin/vagrant_provision:71]: That looks like a good option :) Can you make a new ticket for it?

**Pull request merged**